internal/cloud/awscloud/secure-instance.go

```diff
@@ -0,0 +1,378 @@
+package awscloud
+
+import (
+	"fmt"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/ec2"
+	"github.com/sirupsen/logrus"
+)
+
+type SecureInstance struct {
+	FleetID  string
+	SGID     string
+	LTID     string
+	Instance *ec2.Instance
+}
+
+func (a *AWS) RunSecureInstance(iamProfile string) (*SecureInstance, error) {
+	identity, err := a.ec2md.GetInstanceIdentityDocument()
+	if err != nil {
+		logrus.Errorf("Error getting the identity document, %s", err)
+		return nil, err
+	}
+
+	descrInstancesOutput, err := a.ec2.DescribeInstances(&ec2.DescribeInstancesInput{
+		InstanceIds: []*string{
+			aws.String(identity.InstanceID),
+		},
+	})
+	if err != nil {
+		return nil, err
+	}
+	if len(descrInstancesOutput.Reservations) != 1 || len(descrInstancesOutput.Reservations[0].Instances) != 1 {
+		return nil, fmt.Errorf("Expected exactly one instance")
+	}
+	vpcID := *descrInstancesOutput.Reservations[0].Instances[0].VpcId
+	imageID := *descrInstancesOutput.Reservations[0].Instances[0].ImageId
+
+	var secureInstance *SecureInstance
+	defer func() {
+		if secureInstance != nil && secureInstance.Instance == nil {
```

Comment:
This is pretty long for a defer :) I think it would be better as a cleanup function that's called. Or at least turn the wrapping nil checks into an early return so the actual code can be indented less.